test/integration/services/data/test-get-staging-case-details.js

```diff
@@ -0,0 +1,27 @@
+
+const expect = require('chai').expect
+const getStagingCaseDetails = require('../../../../app/services/data/get-staging-case-details')
+const caseDetailsHelper = require('../../../helpers/data/staging-case-details-helper')
+
+describe('services/data/get-staging-case-details', function () {
+  before(function () {
+    return caseDetailsHelper.insertOverdueTermination().then(function () {
+      return caseDetailsHelper.insertPriority().then(function () {
+        return caseDetailsHelper.insertUnpaidWork().then(function () {
+          return caseDetailsHelper.insertWarrant()
+        })
+      })
+    })
+  })
+
+  it('should return the union of all staged case details', function () {
+    return getStagingCaseDetails()
+      .then(function (caseDetails) {
+        expect(caseDetails.length).to.be.equal(4)
+      })
+  })
+
+  after(function () {
+    caseDetailsHelper.deleteAll()
```

Comment:
One thing to note about this is that there is no ID being set to tie data created specifically in the `before` to what we need to clean up. So the tests are a bit leaky - will need to fix this